webpack.config.js

```diff
@@ -22,12 +24,19 @@ const conf = {
     libraryTarget: 'umd',
     path: path.resolve(__dirname, 'dist')
   },
-  plugins: []
+  plugins: [
+    new BundleAnalyzerPlugin({
+      analyzerMode: 'static',
+      openAnalyzer: false
+    })
+  ]
 };
 
 if (ENVIRONMENT === 'production') {
-  conf.externals.react = 'react';
-  conf.externals['react-dom'] = 'react-dom';
+  conf.externals.push('react');
+  Object.keys(pkg.dependencies).forEach((dep) => {
```

Comment:
this will also push mediaquery as an external dependency in production, while it is actually needed for the library to work. Or amI missing something?

Reply:
Yes, it will push mediaquery as an external. Dependencies don’t need to be in dist/rezposive.js if the users of this lib use webpack, browserify, etc. because those bundlers can resolve and possibly dedupe dependencies. However, this will break usecases with AMD or global namespace. I am not sure whether people are using React in those ways though.

Now I think that it’s better to add transpiled JS files for the CommonJS usecase and leave the UMD build as is, like http://dev.topheman.com/package-a-module-for-npm-in-commonjs-es2015-umd-with-babel-and-rollup/ In this way, webpack users can dedupe dependencies and save some bytes of webpack helpers that this lib’s webpack generated.

What do you think?